backend/handler/metadata/sgdb_handler.py

```diff
@@ -62,28 +133,19 @@ async def get_details_by_names(self, game_names: list[str]) -> SGDBRom:
                 log.debug(f"Could not find '{search_term}' on SteamGridDB")
                 continue
 
-            # SGDB search is fuzzy so no need to split the search term by special characters
-            search_term_normalized = self.normalize_search_term(
-                search_term, remove_articles=False
-            )
-
-            # Calculate Levenshtein distances for all games and find the best match
-            game_distances = []
+            game_scores = []
             for game in games:
-                game_name_normalized = self.normalize_search_term(
-                    game["name"], remove_articles=False
-                )
-                distance = levenshtein_distance(
-                    game_name_normalized, search_term_normalized
+                similarity_score = self._calculate_title_similarity(
+                    search_term, game["name"]
                 )
-                game_distances.append((game, distance))
+                game_scores.append((game, similarity_score))
```

Comment:
This could be in a separate PR, but getting a similarity score of `1.0` could trigger a break here, so there's no need to keep calculating for the rest of the list.